### ISSUE TITLE
[Filebeat] Add grok Processor as native beat/filebeat processor 

### ISSUE THREAD

**Issue by mr1716** [elastic/beats#30073]
This enhancement request is to make the grok processor that is native in ElasticSearch Ingest Processors to become native in a future release of Beats, such as Filebeat. Ideally, this can be done in a version 8.X release.

There is already the dissect processor in Filebeat, and other Beats, and by adding a grok processor it would help keep maintain consistency, and speed up processing on large scale deployments. There are some gaps that Grok would cover that the Beats platform is missing with the Dissect Processor. It is the 8th most common processor, seen in 38 of 70 modules in Filebeat.

**Comment by mr1716** [comment]
Team: <enhancement>

**Comment by mr1716** [comment]
Would it be possible to have the Filebeat and Enhancement tags attached to this as well please?  @mtojek

**Comment by UcanInfosec** [comment]
It would be great if this could be added! Really useful.

**Comment by mr1716** [comment]
@jlind23 @nimarezainia @woodywalton is there any way that this could get prioritized or have more visibility?

**Comment by jlind23** [comment]
@mr1716 I defer to Nima here but so far it hasn't been prioritised as we are handling higher priority with the team. I am afraid that nothing would change here until at least Q2 2023.

**Comment by UcanInfosec** [comment]
@jlind23 thanks for the heads up. Adding this feature would be great!

**Comment by nimarezainia** [comment]
@UcanInfosec @mr1716 As noted the grok processor is available in the ingest pipelines and can be applied there consistently to all ingress data (from all the beats). Maintaining processors consistently on all the beats is challenging for users and leads to issues.

Is there a reason why you can't use the ingest pipeline to do this?

At this point this effort is not on our list of priorities.

**Comment by UcanInfosec** [comment]
@nimarezainia at this time, it isn’t a priority to implement ingest pipelines and will lead to issues and challenges with the use of elastic tools. If it’s too much to implement, it might be easier for us to switch to a competitor going forward. And this should be noted for those who want this functionality and are not using an elastic tool supporting ingest pipelines

**Comment by mr1716** [comment]
@jlind23 Hey. I know that this was pushed back to at least Q2 of 2023. Is there any more sense of timeline?

**Comment by nimarezainia** [comment]
@mr1716 apologies this is currently not part of our development backlog due to many other priorities on hand at the moment.

**Comment by mr1716** [comment]
@nimarezainia thanks. Will stsrt evaluating a new tool then that might better suit the needs to replace filebeat/elastic integration agent.

**Comment by james-mchugh** [comment]
@nimarezainia Would it be possible for me (or another contributor) to submit a PR for this?

**Comment by nimarezainia** [comment]
@james-mchugh yes absolutely. Let me know and we will review it for you. thanks.

**Comment by james-mchugh** [comment]
Awesome. Thank you! It may take me a bit to get something out, but I'll start working on it. As recommended in the contributor guidelines, I created a discussion here to ask the community for any recommendations https://discuss.elastic.co/t/beats-native-grok-processor/336521.

**Comment by studyhuang1996** [comment]
This function is very useful, when the target side is not es, and do not want to add a layer of logstash processing, grok is particularly important, need to officially support this feature, so that filebeat can be a very independent individual, the scope of application is very wide, fluent-bit also supports multiple lines, otherwise when parsing text in multiple lines, the use of the current dissect for some scenarios can not be supported

**Comment by zez3** [comment]
still valid

**Comment by rpasche** [comment]
Hi @james-mchugh , I just found https://pkg.go.dev/github.com/elastic/go-grok, that looks pretty promising....

**Comment by rpasche** [comment]
Wow... I started to look into this on my own today....and with help of some AI, I seem to have a first *basic* version working. I will try to add some "tests" as well....or at least, will try to file a PR, to get some feedback and improvement suggestions.

**Comment by strawgate** [comment]
We’re on a path to integrate OTel-based collection natively into the Elastic ingestion architecture.

The plan is to allow running OTel receivers, Beats, and Elastic Integrations side-by-side within a single collector. We'll have additional information about this coming in the near future.

For this reason we won't be introducing a Grok processor into Beats as this time.

In the meantime, can you review the [Grok OTTL Statement](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/ottl/ottlfuncs#extractgrokpatterns) we contributed to OpenTelemetry and let us know what features are missing that you would need it to have in order to meet your use-case?
